pinc/stages.inc

```diff
@@ -11,360 +11,537 @@ include_once($relPath.'Pool.inc');
+            _("If nothing interests you right now, check back later and there will be more!"),
+            "</p>",
+
+            "<p>",
+            _("Each book listed below has gone through multiple rounds of proofreading and formatting, and now needs to be massaged into a final e-text."),
+            _("Once you have checked out and downloaded a book it will remain checked out to you until you check it back in."),
+            _("When you have finished your work on the book, select <i>Upload for Verification</i> from the drop-down list for that project."),
+            _("If you have several files to submit for a single project (say a text and HTML version), zip them up together first."),
+            "</p>",
+        ]
+    ));
+
+    Activities::set(new Pool(
+        'PPV',
+        _('Post-Processing Verification'),
+        [],
+        'NOREQ', // "Peer approval. Also gives F2 access."
+        '',
+        null, // access_change_callback
+        _('Once a PPer has submitted a final e-text, it needs to be checked by a PPVer before it is posted to PG.'),
+        'ppv.php',
+        _("Post-Processor"),
+        'postproofer',
+        [
+            "<p>",
+            _("In this pool, experienced volunteers verify texts that have already been Post-Processed, and mentor new Post-Processors."),
+            sprintf(
+                _("<b>Before working in this pool</b>, please make sure you read the <a href='%s'>Post-Processing Verification Guidelines</a>."),
+                "$code_url/faq/ppv.php"
+            ),
+            sprintf(
+                _("The PPV reporting form is <a href='%s'>here</a>."),
+                "$code_url/tools/post_proofers/ppv_report.php"
+            ),
+            "</p>",
+
+            "<p>",
+            sprintf(
+                _("As always, the <a href='%s'>Post-Processing Forum</a> is available for any of your questions."),
+                get_url_to_view_forum($post_processing_forum_idx)
+            ),
+            "</p>",
+        ]
+    ));
+
+    Activities::set(new Stage(
+        'SR',
+        _('Smooth Reading'),
+        [],
+        'IMMEDIATE',
+        '',
+        null, // access_change_callback
+        _('Before a final e-text is posted to PG, it can be optionally uploaded for Smooth Reading. Anyone can volunteer to Smooth Read a text, which involves reading through the text for smoothness, marking possible errors and returning it to the PPer.'),
+        null,
+        "tools/post_proofers/smooth_reading.php"
+    ));
+
+    Activities::set(new Activity(
+        'DU',
+        _('Uploading to Project Gutenberg'),
+        [],
+        'NOREQ',
+        'OOGA',
+        null // access_change_callback
+    ));
+
+    Activities::set(new Activity(
+        "P2_mentor",
+        sprintf(_("Mentoring in round %s"), 'P2'),
+        [],
+        'NOREQ',
+        '',
+        null // access_change_callback
+    ));
+
+    // now construct various access tables
```

Comment:
"tables" isn't quite the right word. I believe you're thinking of "lookup tables" but in PHP those are just associative arrays (in python they're called dictionaries). "tables" in our code with the DB backend really implies "database table" and I'd encourage us to avoid using "table" for the array.

Reply:
Yes, I did occur to me earlier that what this stuff does is like creating a database in memory of all the activity information with Key lookup tables to find the information in various ways. I'll change that comment.